nupic/research/frameworks/pytorch/imagenet/imagenet_experiment.py

```diff
@@ -330,14 +333,18 @@ def create_validation_dataloader(cls, config):
             num_classes=config.get("num_classes", 1000),
         )
 
-        # Non-distributed.
+        if config.get("distributed", False):
+            sampler = UnpaddedDistributedSampler(dataset, shuffle=False)
+        else:
+            sampler = None
         return DataLoader(
             dataset=dataset,
             batch_size=config.get("val_batch_size",
                                   config.get("batch_size", 1)),
             shuffle=False,
             num_workers=config.get("workers", 0),
-            pin_memory=False,
+            sampler=sampler,
+            pin_memory=True,
```

Comment:
`pin_memory` only works on cuda. It is safer to use `pin_memory=torch.cuda.is_available()`